src/dft/backends/rocfft/commit.cpp

```diff
@@ -259,12 +260,37 @@ class rocfft_commit final : public dft::detail::commit_impl<prec, dom> {
         std::reverse(stride_vecs.vec_b.begin(), stride_vecs.vec_b.end());
         stride_vecs.vec_b.pop_back(); // Offset is not included.
 
-        rocfft_plan_description plan_desc;
-        if (rocfft_plan_description_create(&plan_desc) != rocfft_status_success) {
+        // This workaround is needed due to a confirmed issue in rocFFT from version
+        // 1.0.23 to 1.0.30. Those rocFFT version correspond to rocm version from
+        // 5.6.0 to 6.3.0.
+        // Link to rocFFT issue: https://github.com/ROCm/rocFFT/issues/507
+        if constexpr (rocfft_version_major == 1 && rocfft_version_minor == 0 &&
+                      (rocfft_version_patch > 22 && rocfft_version_patch < 31)) {
+            if (dom == dft::domain::COMPLEX && dimensions > 2) {
+                auto stride_checker = [&](const auto& a, const auto& b) {
+                    for (ulong i = 0; i < dimensions; ++i) {
+                        if (a[i] != b[i])
+                            return false;
+                    }
+                    return true;
+                };
+                std::printf("hello\n");
+                if (!stride_checker(stride_vecs.vec_a, stride_vecs.vec_b))
```

Comment:
Can't you use `stride_vecs.vec_a != stride_vecs.vec_b`? Or is `dimensions` smaller than the length of these vectors?

Reply:
Honestly, I did not check. Dimensions should be the same. I'll try and update it.